liiatools/datasets/s903/lds_ssda903_clean/file_creator.py

```diff
@@ -26,7 +26,7 @@ def coalesce_row(stream):
         elif isinstance(event, events.EndRow):
             yield RowEvent.from_event(event, row=row)
             row = None
-        elif row is not None and isinstance(event, events.Cell):
+        elif row is not None and isinstance(event, events.Cell) and event.header in set(event.expected_columns):
```

Comment:
For example, if I put the following into python, I get an error:

`['name'] in {'address', 'name', 'DOB'}`

`TypeError: unhashable type: 'list'`

If event.header is a tuple or a set, however, then you're fine.

Reply:
event.header (an attribute of event.cell) will always just be a single item. However, there is also event.headers (an attribute of event.StartTable) which is a set of these. I think this may be the source of your confusion as it's the latter which is called as a set in the configuration file.

Reply:
I understand. I just could see that the attributes were a bit inconsistent and wanted to challenge that a bit. If you're happy that things are fine, then I'm happy to approve.